pkg/sql/schemachanger/comparator_generated_test.go

```diff
@@ -0,0 +1,14316 @@
+
+	var path string
+	if bazel.BuiltWithBazel() {
+		var err error
+		path, err = bazel.Runfile(logicTestFile)
+		require.NoError(t, err)
+	} else {
+		path = "../../../" + logicTestFile
+	}
+	basename := filepath.Base(path)
+
+	var skipComparatorTest = envutil.EnvOrDefaultBool("COCKROACH_SCHEMA_CHANGE_COMPARATOR_SKIP", true)
+	if skipComparatorTest {
+		skip.IgnoreLint(t, "requires COCKROACH_SCHEMA_CHANGE_COMPARATOR_SKIP to be set")
+	}
+
+	if skipEntry, skipReason := shouldSkipLogicTestCorpusEntry(basename); skipEntry {
+		skip.IgnoreLint(t, skipReason)
+	}
+
+	stmts, err := collectStmtsFrom(path)
+	require.NoError(t, err)
+
+	ss := &staticSQLStmtLineProvider{
+		stmts: stmts,
+	}
+	sctest.CompareLegacyAndDeclarative(t, ss)
+}
+func TestSchemaChangeComparator_rename_table(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	var logicTestFile = "pkg/sql/logictest/testdata/logic_test/rename_table"
+
+	var path string
+	if bazel.BuiltWithBazel() {
+		var err error
+		path, err = bazel.Runfile(logicTestFile)
+		require.NoError(t, err)
+	} else {
+		path = "../../../" + logicTestFile
+	}
+	basename := filepath.Base(path)
+
+	var skipComparatorTest = envutil.EnvOrDefaultBool("COCKROACH_SCHEMA_CHANGE_COMPARATOR_SKIP", true)
+	if skipComparatorTest {
+		skip.IgnoreLint(t, "requires COCKROACH_SCHEMA_CHANGE_COMPARATOR_SKIP to be set")
+	}
+
+	if skipEntry, skipReason := shouldSkipLogicTestCorpusEntry(basename); skipEntry {
+		skip.IgnoreLint(t, skipReason)
+	}
+
+	stmts, err := collectStmtsFrom(path)
+	require.NoError(t, err)
+
+	ss := &staticSQLStmtLineProvider{
+		stmts: stmts,
+	}
+	sctest.CompareLegacyAndDeclarative(t, ss)
+}
+func TestSchemaChangeComparator_rename_view(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	var logicTestFile = "pkg/sql/logictest/testdata/logic_test/rename_view"
+
+	var path string
+	if bazel.BuiltWithBazel() {
+		var err error
+		path, err = bazel.Runfile(logicTestFile)
+		require.NoError(t, err)
+	} else {
+		path = "../../../" + logicTestFile
+	}
+	basename := filepath.Base(path)
+
+	var skipComparatorTest = envutil.EnvOrDefaultBool("COCKROACH_SCHEMA_CHANGE_COMPARATOR_SKIP", true)
+	if skipComparatorTest {
+		skip.IgnoreLint(t, "requires COCKROACH_SCHEMA_CHANGE_COMPARATOR_SKIP to be set")
+	}
+
+	if skipEntry, skipReason := shouldSkipLogicTestCorpusEntry(basename); skipEntry {
+		skip.IgnoreLint(t, skipReason)
+	}
+
+	stmts, err := collectStmtsFrom(path)
+	require.NoError(t, err)
+
+	ss := &staticSQLStmtLineProvider{
+		stmts: stmts,
+	}
+	sctest.CompareLegacyAndDeclarative(t, ss)
+}
+func TestSchemaChangeComparator_reset(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	var logicTestFile = "pkg/sql/logictest/testdata/logic_test/reset"
+
+	var path string
+	if bazel.BuiltWithBazel() {
+		var err error
+		path, err = bazel.Runfile(logicTestFile)
+		require.NoError(t, err)
+	} else {
+		path = "../../../" + logicTestFile
+	}
+
```

Comment:
So, first of all: you do not have to check this file in. Feel free to let it be `gitignore`'d. If you are going to check this in, I would encourage you to refactor out the massive duplication just so this file doesn't have have to be as colossal as it is.

Reply:
i have removed the massive duplication

Reply:
Oh, awesome, thank you. Now it just slows down my browser instead of crashing it. 😛